libs/langchain/langchain/chat_models/llama2.py

```diff
@@ -0,0 +1,123 @@
+"""Wrapper for llama-2-*-chat LLMs."""
+from typing import Any, List, Optional
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain.chat_models.base import BaseChatModel
+from langchain.llms.base import LLM
+from langchain.schema import (
+    AIMessage,
+    BaseMessage,
+    ChatGeneration,
+    ChatResult,
+    HumanMessage,
+    LLMResult,
+    SystemMessage,
+)
+
+B_INST, E_INST = "[INST]", "[/INST]"
```

Comment:
Nice to see that these tokens are now officially recommended, [as you point out](https://huggingface.co/blog/llama2#how-to-prompt-llama-2).

I was [wondering](https://twitter.com/RLanceMartin/status/1681879318493003776?s=20) whether they were going to get broadly suggested.

Reply:
Inspiring to see how far GPT4 gets deriving a llama2 prompt from analyzing `generation.py`. Looks good except from the wrongly placed first `[INST]`.